### ISSUE TITLE
A suggestion to help speed up guesser/dan.py

### ISSUE THREAD

**Issue by dhawaljoh** [miyyer/qb#9]
In the guesser/dan.py file list 174.
`r = r - update`

Changing that to `r -= update` should help speed things up a tiny bit.

**Comment by miyyer** [comment]
thanks, i made the change. the real bottleneck of the guesser is in line 62 of guesser/classify/learn_classifiers.py, which trains an sklearn multiclass logistic regression (one vs all approach, separate binary classifier for every answer). this takes 2.5 days or so, and while we can dramatically speed it up by replacing the one v all with a softmax layer/cross entropy loss (a theano implementation takes minutes to complete training), accuracy seems to greatly suffer as a result. we'll be looking into alternatives to one vs all for future versions of the system, and any suggestions are welcome :)

**Comment by dhawaljoh** [comment]
I just happened to navigate here after watching JBG's QANTA vs Ken video on YouTube.
That was a small fix I just happened to see.

I shall try to make a more meaningful contribution after I'm more familiar with the code.
